db/collection.go

```diff
@@ -874,7 +874,15 @@ func (c *collection) save(
 		}
 
 		if val.IsDirty() {
-			fieldKey, fieldExists := c.tryGetFieldKey(primaryKey, k)
+			fieldKey, isAliasKey, fieldExists := c.tryGetFieldKey(primaryKey, k)
+			// Overwrite the aliased key with the internal related object name.
+			if isAliasKey {
+				oldKey := k
+				k = oldKey + request.RelatedObjectID
+				doc.Fields()[k] = v
+				delete(doc.Fields(), oldKey)
```

Comment:
tip-for-reviewer: This implementation and some other parts or this commit were removed and implemented elsewhere while fixing a bug in the commit [`PR(FIX): Squash the dockey verification + diff bug`]